h/services/user_signup.py

```diff
@@ -80,7 +86,23 @@ def signup(self, require_activation=True, **kwargs):
 
         # Create a new activation for the user
         if require_activation:
-            self._require_activation(user)
+            try:
+                self._require_activation(user)
+            except IntegrityError as e:
+                if (
+                    'duplicate key value violates unique constraint "uq__user__email"'
+                    in e.args[0]
+                    or 'duplicate key value violates unique constraint "ix__user__userid"'
+                    in e.args[0]
+                ):
+                    log.warning("email conflict error occured during user signup")
```

Comment:
It would be useful to know which of the possible conflicts occurred if this happens. At the very least I think it would be useful for the log message to indicate the uncertainty ("email or username conflict error...") to avoid confusion when someone examines the log messages.